### PR TITLE
Enable specific entities on buoyancy plugin

### PR DESCRIPTION
**Pull request by chapulina**
# 🦟 Bug fix

This is technically a feature, but it fixes a usability bug that affects the Fortress demo.

## Summary

Adds the ability of whitelisting just a few models in simulation which will be affected by buoyancy.

Also reduced console spam by printing a warning only once.

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [x] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸

### CODE REVIEW

File: examples/worlds/graded_buoyancy.sdf [hunk 1]
```diff
@@ -33,75 +33,13 @@
           <density>1</density>
         </density_change>
       </graded_buoyancy>
+      <white_list>ball_lighter_than_water</white_list>
```
Comment (azeey) [inline examples/worlds/graded_buoyancy.sdf:36]: It's not clear if this refers to models or links. Can this be `enable_model` instead?

Comment (chapulina) [reply examples/worlds/graded_buoyancy.sdf:36]: Ah I meant for it to work with either the top level model or the link, as long as the scoped name is given. Nested models would be a bit more work. Let me improve this.

Comment (chapulina) [reply examples/worlds/graded_buoyancy.sdf]: Improved logic to support scoped names of models, links and nested models in 
fe6f894

**Review by arjo129 — APPROVED** [review]
Tested this. It seems to work for me. There are some style fixes required before merging though.

File: src/systems/buoyancy/Buoyancy.cc [hunk 2]
```diff
@@ -121,6 +122,10 @@ class ignition::gazebo::systems::BuoyancyPrivate
   /// at _pose.
   public: std::pair<math::Vector3d, math::Vector3d> ResolveForces(
     const math::Pose3d &_pose);
+
+  /// \brief Scoped names of entities that buoyancy should apply to. If empty,
+  /// all links will receive buoyancy.
+  public: std::unordered_set<std::string> whiteList;
```
Comment (arjo129) [inline src/systems/buoyancy/Buoyancy.cc:128]: Add an `#include <unordered_set>` to make include what you use happy...

File: examples/worlds/graded_buoyancy.sdf (hunk 1 above)
Comment (azeey) [reply examples/worlds/graded_buoyancy.sdf]: Ok, I'm also not a fan of using `whilte_list` and `black_list`. How about calling it `allow_list` or `enable_list` or something along those lines?

Comment (arjo129) [reply examples/worlds/graded_buoyancy.sdf]: Thats actually a good point... I think just `<enable>` could work as well.

Comment (chapulina) [reply examples/worlds/graded_buoyancy.sdf]: Ouch, you're right! Changed to `enable` in 6a83fac

File: examples/worlds/graded_buoyancy.sdf [hunk 3]
```diff
@@ -33,74 +33,18 @@
...
-          <property type="double" key="z">1</property>
 
-          <property type="string" key="state">floating</property>
-          <anchors target="3D View">
-            <line own="left" target="left"/>
-            <line own="bottom" target="bottom"/>
-          </anchors>
-        </ignition-gui>
+      <!-- whitelist by link name -->
+      <white_list>lighter_than_water::ball::body</white_list>
```
Comment (azeey) [inline examples/worlds/graded_buoyancy.sdf:38]: `whilte_list` -> `enable`

File: test/worlds/graded_buoyancy.sdf [hunk 4]
```diff
@@ -33,6 +33,15 @@
           <density>1</density>
         </density_change>
       </graded_buoyancy>
+
+      <!-- whitelist by link name -->
+      <white_list>lighter_than_water::ball::body</white_list>
```
Comment (azeey) [inline test/worlds/graded_buoyancy.sdf:38]: `whilte_list` -> `enable`

Should the test be failing without this change?

Comment (chapulina) [reply test/worlds/graded_buoyancy.sdf]: Ah good to catch this, the `no_buoyancy` model was too dense and would sync either way. Fixed in 5b9bc46

**Comment by chapulina** [comment]
I've addressed all comments, merging!

**Pull request merged**